torch/csrc/jit/python/init.cpp

```diff
@@ -1398,79 +1601,152 @@ void initJITBindings(PyObject* module) {
               })
           .def(
               "__add__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return a->add(snb);
-              })
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->add(snb));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->add(snb));
+                } else {
+                  throw py::reference_cast_error();
+                }
+              }, py::is_operator())
           .def(
               "__radd__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return snb->add(a);
-              })
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(snb->add(a));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(snb->add(a));
+                } else {
+                  throw py::reference_cast_error();
+                }
+              }, py::is_operator())
           .def(
               "__sub__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return a->sub(snb);
-              })
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->sub(snb));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->sub(snb));
+                } else {
+                  throw py::reference_cast_error();
```

Comment:
It looks like this fixes `symint.add(Tensor)` (we want that call to return `NotImplemented`, so python will try `Tensor.radd(symint)`). For some reason pybind wasn't doing that for `py::cast()` failures.